### PR TITLE
Upgrading Azure modules

### PR DESCRIPTION
**Pull request by oxisto**
It seems that for compute and App-services, new major module versions exist, which were not picked up by @dependabot.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -165,7 +164,7 @@ require (
 	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
 	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
-	sigs.k8s.io/yaml v1.3.0 // indirect
+	sigs.k8s.io/yaml v1.2.0 // indirect
```
Comment (lebogg) [inline go.mod:167]: Isn't it a downgrade?

**Review by anatheka — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (oxisto) [reply go.mod:167]: 🤷 wasn't me. This was introduced by `go tidy`.

**Review by lebogg — APPROVED** [review]

File: service/discovery/azure/compute.go [hunk 2]
```diff
@@ -30,8 +30,8 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/appservice/armappservice"
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/appservice/armappservice/v2"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v3"
```
[the remark this answers is not in the document]
Comment (oxisto) [reply service/discovery/azure/compute.go:34]: We should use the latest major version and therefore, we need the version string

File: service/discovery/azure/storage.go [hunk 3]
```diff
@@ -35,12 +35,13 @@ import (
 	"clouditor.io/clouditor/internal/util"
 	"clouditor.io/clouditor/voc"
 
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v3"
```
[the remark this answers is not in the document]
Comment (oxisto) [reply service/discovery/azure/storage.go:38]: We should use the latest major version and therefore, we need the version string

**Pull request merged**